### ISSUE TITLE
twisted/conch/test/test_filetransfer.py has fragile setUp/tearDown

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1788]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| spiv reported|
|-|-|
|Trac ID|trac#1788|
|Type|enhancement|
|Created|2006-06-04 15:11:43Z|

http://twistedmatrix.com/buildbot/reactors/builds/1363/step-gtk/2 has some odd failures in test_filetransfer.py that look like they may be due to inadequate test isolation -- the test directory has garbage that looks like it's left over from another test.

I'm attaching a patch that simplifies test setUp and tearDown in this file, as well as improving the isolation.

Attachments:

* [conch-test-cleanup.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/cf7/cf70b1cc3fb8c74f6faf74b0d1ca03130b179c90/dc2657f6c99779bf7ad3010d1e54c365d9038960.diff) (4323 bytes) - added by spiv on 2006-06-04 15:12:20Z - 
* [conch-test-cleanup-2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/cf7/cf70b1cc3fb8c74f6faf74b0d1ca03130b179c90/b3bd47b85b69d12cbe62d6b18369267b54426f5c.diff) (4265 bytes) - added by spiv on 2006-06-04 15:21:49Z - 
* [conch-fixes.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/cf7/cf70b1cc3fb8c74f6faf74b0d1ca03130b179c90/e52278e5d22cecd9b6a807ae70c98c1becb54e5e.diff) (9233 bytes) - added by spiv on 2006-06-04 17:18:41Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1788 1788
type__enhancement enhancement
reporter__spiv spiv
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1149433903000000 1149433903000000
changetime__1149518021000000 1149518021000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

Specifically, it looks tests in test_filetransfer.py are being victimised by twisted.conch.test.test_cftp.TestOurServerCmdLineClient.testGet timing out, because they currently share the 'sftp_test' directory, *and* the current setUp in test_filetransfer doesn't take any steps to cope if that directory exists and has garbage in it.  This patch uses the mktemp feature of trial to give every test method in test_filetransfer its own directory, so tests won't interfere with each other by accident.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

exarkun points out that the tearDown is now unnecessary, so I've removed it entirely.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to spiv|
|-|-|

I can't get these to fail locally without the patch, but they still all pass after the patch too.  And the patch looks pretty good.  One more change I'd suggest - pyflakes says shutil, protocol, interfaces, and failure are imported but unused.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set status to `closed`|
|-|-|

(In [17063]) Simplify test setUp in test_filetransfer, also improving test isolation.

 * Fixes [#1788](https://github.com/twisted/twisted/issues/1788)
 * Author: spiv
 * Reviewer: exarkun

This prevents failures in tests like 
twisted.conch.test.test_cftp.TestOurServerCmdLineClient.testGet from 
victimising tests in test_filetransfer.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set status to `reopened`|
|-|-|

Turns out test_cftp is affected by that patch -- here's most of the fix for that.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set owner to @exarkun<br>spiv set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

exarkun has made [branches/conch-test-fixes-1788](branches/conch-test-fixes-1788) for this work.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to spiv|
|-|-|

Aside from underscore variables, looks good (again ;).  Tests pass on my machine and on several builders.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set status to `closed`|
|-|-|

(In [17083]) Simplify test setUp in test_filetransfer, while also improving test isolation.

 * Fixes [#1788](https://github.com/twisted/twisted/issues/1788)
 * Author: spiv
 * Reviewer: exarkun

This prevents failures in tests like
twisted.conch.test.test_cftp.TestOurServerCmdLineClient.testGet from
victimising other tests in test_cftp and test_filetransfer.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
